### PR TITLE
8-bit strings

### PR DESCRIPTION
**Pull request by giann**

### CODE REVIEW

File: src/lparser.js [hunk 1]
```diff
@@ -405,7 +405,7 @@ const findlabel = function(ls, g) {
     /* check labels in current block for a match */
     for (let i = bl.firstlabel; i < dyd.label.n; i++) {
         let lb = dyd.label.arr[i];
-        if (lb.name.value === gt.name.value) {  /* correct label? */
+        if (lb.name.jsstring() === gt.name.jsstring()) {  /* correct label? */
```
Comment (daurnimator) [inline src/lparser.js]: Why can't you just compare values? Would they be the same due to interning?

File: src/lstate.js [hunk 2]
```diff
@@ -69,14 +69,23 @@ class global_State {
 
     constructor(L) {
         this.mainthread = L;
-        this.strt = null; // TODO: string hash table
+        this.strt = new Map();
```
Comment (daurnimator) [inline src/lstate.js]: How does this get cleared? Wouldn't it just grow for every string created?

Comment (giann) [inline src/lstate.js]: It will yes. I wrote a really simple and naive to begin with. I'm open to suggestions.

File: src/lparser.js (hunk 1 above)
Comment (giann) [inline src/lparser.js]: Yes, did not think of that

File: src/lstate.js (hunk 2 above)
Comment (daurnimator) [inline src/lstate.js]: I wouldn't implement a global string table at all: I'd just do it at the point of comparison/set-in-table.

Additionally I was contemplating having two separate string implementations: for "normal" strings (i.e. 7-bit strings), just use normal JS strings. If they go multibyte, use a `UInt8Array`

Comment (giann) [inline src/lstate.js]: What would the benefits be ?

**Pull request merged**